File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/structure/DepartmentDto.java
package org.burningokr.dto.structure;

import java.util.ArrayList;
import java.util.Collection;
import java.util.UUID;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class DepartmentDto extends SubStructureDto {

  public DepartmentDto() {
    super(StructureType.DEPARTMENT);
  }

  private UUID okrMasterId;

  private UUID okrTopicSponsorId;

  private Collection<UUID> okrMemberIds = new ArrayList<>();
}

File: frontend/src/app/shared/model/api/structure/department.dto.ts
import { SubStructureDto } from './sub-structure.dto';

export class DepartmentDto extends SubStructureDto {
  okrMasterId: string;
  okrTopicSponsorId: string;
  okrMemberIds: string[];
  isParentStructureACorporateObjectiveStructure?: boolean;
}
